### DISCUSSION TITLE
Reset Page Counter doesn't work when `number-align: top` is used

### DISCUSSION THREAD

**Discussion by HiramTheHero** [typst/typst#2831] [General]
### Description

Within the `counter` page in the Typst docs it gives the following example to demonstrate how to reset page numbering:
```
#set page(numbering: "(i)")

= Preface
The preface is numbered with
roman numerals.

#set page(numbering: "1 / 1")
#counter(page).update(1)

= Main text
Here, the counter is reset to one.
We also display both the current
page and total number of pages in
Arabic numbers.
```

And works as expected. However, when page numbering is set to the top of the page, the counter isn't reset properly. The following code causes the second page to have 2 / 1 as the page numbering instead of 1 / 1:
```
#set page(numbering: "(i)", number-align: top)

= Preface
The preface is numbered with
roman numerals.

#set page(numbering: "1 / 1")
#counter(page).update(1)

= Main text
Here, the counter is reset to one.
We also display both the current
page and total number of pages in
Arabic numbers.
```

### Operating system

macOS

### Typst version

- [X] I am using the latest version of Typst

**Comment by Enivex** [comment]
This is working as intended. You're updating the page counter _after_ you've started a new page. That means that on the second page, the counter is 2, but the final value of the counter is 1 since that's what you're changing it to.

The correct way to achieve the desired effect is
```
#counter(page).update(0)
#set page(numbering: "1 / 1")
```

**Comment by HiramTheHero** [comment]
Thank you for the help! Sorry for the silly error.
